org.pgadmin.pgadmin4.yml

```diff
@@ -0,0 +1,146 @@
+app-id: org.pgadmin.pgadmin4
+runtime: org.freedesktop.Platform
+runtime-version: "24.08"
+base: org.electronjs.Electron2.BaseApp
+base-version: "24.08"
+sdk: org.freedesktop.Sdk
+command: run.sh
+separate-locales: false
+rename-desktop-file: pgadmin4.desktop
+rename-icon: pgadmin4
+finish-args:
+  - --share=ipc
+  - --share=network
+  - --socket=x11
+  - --socket=pulseaudio
+  - --device=dri
+  - --talk-name=org.freedesktop.secrets
+modules:
+  - name: libpq
+    # Copying build instructions for the libpg client library from
+    # https://src.fedoraproject.org/rpms/libpq/blob/rawhide/f/libpq.spec
+    config-opts:
+      - --disable-rpath
+      #- --with-ldap  # Disabling until somebody actually needs it
+      - --with-openssl
+      #- --with-gssapi  # Disabling until somebody actually needs it
+      - --enable-nls
+      - --without-readline
+    sources:
+      - type: git
+        url: https://git.postgresql.org/git/postgresql.git
+        commit: 3edc6580c0e27fb8f13322efd255a88d20dda6c2
+        tag: REL_16_1
+        x-checker-data:
+          type: git
+          tag-pattern: ^REL_([\d_]+)$
+
+  - name: krb5
+    subdir: src
+    config-opts:
+      - --localstatedir=/var/lib
+      - --sbindir=${FLATPAK_DEST}/bin
+      - --disable-static
+      - --disable-rpath
+    sources:
+      - type: archive
+        url: https://kerberos.org/dist/krb5/1.21/krb5-1.21.2.tar.gz
+        sha256: 9560941a9d843c0243a71b17a7ac6fe31c7cebb5bce3983db79e52ae7e850491
+        x-checker-data:
+          type: html
+          url: https://kerberos.org/dist/
+          version-pattern: Kerberos V5 Release ([\d\.-]*) - current release
+          url-template: https://kerberos.org/dist/krb5/$version0.$version1/krb5-$version.tar.gz
+
+  - name: python3-bcrypt
+    buildsystem: simple
+    build-commands:
+      - pip3 install --prefix=/app bcrypt-4.2.0-cp39-abi3-manylinux_2_28_x86_64.whl
```

Comment:
were this generated by hand? Because having the full filename is not correct.

Reply:
this bcrypt entry with url link was provided by hand

Reply:
I'm not sure how this could be solved, without rewriting the flatpak builder tools to do the right thing. Please advice.

Reply:
this was not generated using the tool, the tool does not use filenames. You just have to replace them with `bcrypt`

Reply:
Will do, was hoping you were hinting at a fix in the generation instead.

Just changing it to bcrypt isn't enough. I've been playing with `dest-filename`, but haven't found a working solution yet.

Reply:
> Just changing it to bcrypt isn't enough. I've been playing with `dest-filename`, but haven't found a working solution yet.

Change it to `pip3 install --no-index --find-links=file://${PWD} --prefix=${FLATPAK_DEST} bcrypt`

`--no-index --find-links=file://${PWD}` is important here. That's why I was saying to auto generate it through pip-generator and just replace the `sources` with your wheel.

Reply:
Or ie. keep the build command the same as pip-generator, the full command line to pip is an important detail.